Tests/Services/TasksServiceTests.cs

```diff
@@ -415,4 +408,412 @@ await taskService.SubmitTasksAsync(new SessionInfo("sessionId1"),
+    mockBlobService.SetupCreateBlobMock(expectedBlobs);
+
+    var taskService = MockHelper.GetTasksServiceMock(callInvoker,
+                                                     mockBlobService);
+
+    var taskNodes = new List<TaskNode>
+                    {
+                      new()
+                      {
+                        Payload = new BlobInfo
+                                  {
+                                    BlobName  = "payloadId",
+                                    BlobId    = "blobId",
+                                    SessionId = "sessionId1",
+                                  },
+                        ExpectedOutputs = new List<BlobInfo>
+                                          {
+                                            new()
+                                            {
+                                              BlobName  = "output1",
+                                              BlobId    = "outputId1",
+                                              SessionId = "sessionId1",
+                                            },
+                                          },
+                        Session = new SessionInfo("sessionId1"),
+                        TaskOptions = new TaskConfiguration
+                                      {
+                                        PartitionId = "subtasking",
+                                      },
+                        DataDependenciesContent = new Dictionary<string, ReadOnlyMemory<byte>>
+                                                  {
+                                                    {
+                                                      "dependencyBlob", new ReadOnlyMemory<byte>(new byte[]
+                                                                                                 {
+                                                                                                   1,
+                                                                                                   2,
+                                                                                                   3,
+                                                                                                 })
+                                                    },
+                                                  },
+                      },
+                    };
+
+    var result = await taskService.SubmitTasksAsync(new SessionInfo("sessionId1"),
+                                                    taskNodes);
+
+    ClassicAssert.IsNotNull(result);
+    ClassicAssert.AreEqual("taskId1",
+                           result.First()
+                                 .TaskId);
+    ClassicAssert.AreEqual("payloadId1",
+                           result.First()
+                                 .PayloadId);
+    ClassicAssert.AreEqual("outputId1",
+                           result.First()
+                                 .ExpectedOutputs.First());
```

Comment:
Please use the [Contraint model](https://docs.nunit.org/articles/nunit/writing-tests/assertions/assertions.html) of assertions.